[jannaleong] iP

jannaleong opened the pull request:
# Belle
_Your dream personal todo list assistant_

> "If you dont plan, you plan to fail" 👍 

## Belle can help you in completing your everyday tasks by keep a record of it. Functions include:
* adding tasks to list
* deleting tasks from list
* searching for a list using keyword

## **Easily** download by:
1. click this [link](https://github.com/jannaleong/ip/releases/tag/A-Jar) and download jar file
2. navigate to jar file in terminal
3. run 
```
java -jar duke.jar
```
 ## If you have the following characteristics, you should use Belle.
- [ ] highly busy individual
- [ ] wants to organise their lives

## If you are a java coder, you can use it to practice java too. Below is the `main` method:

```java
public static void main(String[] args) {
        new Belle("src/main/data/belleList.txt").run();
    }
}
```

What more? Belle is ~~expensive~~ **FREE**!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jannaleong committed:
added skeleton greeting (level 0)

jannaleong committed:
mimics user and exits when user inputs bye

jannaleong committed:
adds inputs to list

jannaleong committed:
correction to task level 2 to be able to take in a phrase instead of only words

jannaleong committed:
added task class, can mark and unmark tasks

jannaleong committed:
added subclasses of task and edited duke to exhibit polymorphism

jannaleong committed:
handle exceptions

jannaleong committed:
added delete function

jannaleong committed:
using automated testing

jannaleong committed:
extracted out code into respective methods to make it neater. also added some more catch statements to catch more exceptions.

jannaleong committed:
made some methods in Duke.java private

jannaleong committed:
Add HardDisk.java to save the data from running Duke and reload previous data

Duke has an additional functionality of allowing data to be saved and reloaded.

HardDisk has a loadList and save function that allows data to be reloaded
and saved respectively.

Functions is seperated from the rest of the code within the harddisk class
as they have a common link in which they both utilise stored data

jannaleong committed:
Add DateFormatter.java to convert dates in the format yyyy-mm-dd into a more readable form

This will make the dates more readable for the user.

The DateFormatter class is used in the constructors of EventTask and DeadlineTask to
convert the dates.

This is so a more readable date is displayed to the user.

jannaleong committed:
Restructure files to have more OOP

Everything exists in one file which is messy. Need to seperate file into Ui, Storage, Parser,
TaskList so it is neater.

Seperate code into these different classes. Also seperate commands into classes
that inherit from 1 abstract command class.

jannaleong committed:
Restructure files to place in suitable packages

Without packages, files are messy and classes are hard to find.

Separated into packages command, run, tasks, and others.

With packages workflow is neater.

jannaleong committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

jannaleong committed:
Add Junit test cases for DateFormatter and Parser classes.

This automates testing to make testing faster.

Add Junit tests in the src/java/test folder for greater convenience.

jannaleong committed:
Build jar application

Allow users to conveniently download the app and use.

Change the build.gradle file to set main class as Belle.java.

jannaleong committed:
Add javadocs to public methods.

Able to easily read the functionality of each method.

Added javadocs only to public methods which are not overriding methods
or getters and setters.

This reduces redundant work.

jannaleong committed:
Edit files to fit coding standard.

Without standardising, hard to work in a group with different coding styles.

Follow style guide specified by course.

jannaleong committed:
Add FindCommand class that allows user to find tasks by keyword.

This enables users to find specific tasks faster.

Create a FindCommand class that can print the correct task.

This class maintains OOP principles.

jannaleong committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit files to fit coding standard.

# Conflicts:
#	src/main/java/run/TaskList.java

jannaleong committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add FindCommand class that allows user to find tasks by keyword.

# Conflicts:
#	src/main/data/belleList.txt

jannaleong committed:
Add missing import statement for command.FindCommand.

Does not function without FindCommand Import.

jannaleong committed:
Clean up code : exception handling.

Standardise code to only catch BelleException.

There are inconsistencies across the files where some files catch both IOExceptions and BelleExceptions
and others only catch BelleException. It is messy and hard to keep track of.

jannaleong committed:
Refactor files

Belle.java is not assigned in a package.

Place Belle.java in package duke.

It is a requirement for classes to be in a package.

jannaleong committed:
Add more javadocs.

Only some methods and classes have javadocs.

Add more javadocs so code will be clearer.

jannaleong committed:
Add GUI to project

Gui version of the project has the same functionalities as the text-based ui.

Text-based ui can be difficult for the user to use as the output may be small.
Furthermore, text is in a single colour which is not exciting to the user.

Gui provides a more user-friendly platform for users to interact
with the program.

Let's apply Gui using a simpler method using JavaFx and gradle.

jannaleong committed:
Revised code with checkstyle

Code does not follow the coding standard strictly.

Would lead to inconsistencies in code if the project were
to be done with different members.

Let's apply checkstyle to raise the violations in coding standard.

Using this method, violations are raised efficiently and are fixed.

jannaleong committed:
Add javadocs for main.java file

Start method in main.java file does not have javadocs
to provide a clear explanation of its purpose.

This causes confusion on the methods functionality.

Javadocs would allow other coders and users to understand
the code clearly.

Let's write javadocs for the start method.

jannaleong committed:
Add javadocs for mainwindow.java file

Greetuser method does not have javadocs
to provide a clear explanation of its purpose.

This causes confusion on the methods functionality.

Javadocs would allow other coders and users to understand
the code clearly.

Let's write javadocs for the greetuser method.

jannaleong committed:
Add javadocs for Belle.java file

 Getresponse method does not have javadocs
to provide a clear explanation of its purpose.

This causes confusion on the methods functionality.

Javadocs would allow other coders and users to understand
the code clearly.

Let's write javadocs for the getresponse method.

jannaleong committed:
Add javadocs for addtodo.java file

Execute method does not have javadocs
to provide a clear explanation of its purpose.

This causes confusion on the methods functionality.

Javadocs would allow other coders and users to understand
the code clearly.

Let's write javadocs for the execute method.

jannaleong committed:
Add javadocs for byecommand.java file

execute method does not have javadocs
to provide a clear explanation of its purpose.

This causes confusion on the methods functionality.

Javadocs would allow other coders and users to understand
the code clearly.

Let's write javadocs for the execute method.

Cedricaca commented on src/main/java/duke/tasks/TodoTask.java:
```diff
@@ -0,0 +1,22 @@
+    public String toString() {
+        return " [T]" + super.toString();
+    }
+}
```
Hi, I looked through your code and was quite impressed. There was no coding standard violations as far as I could tell, and I liked that you abstracted the command object which I did not do myself. Good work!

jannaleong replied:
Noted, thank you cedric !!

Cedricaca reviewed (commented):
Looks good! Code is easy to read

jannaleong committed:
Improve Code Quality

Code that does not follow code quality standards is less readable.

To improve readability, follow code quality standards.

Ones to follow include using guard statements, extracting long methods into
smaller ones and using Slap.

jannaleong committed:
Add assertions to code

There are some assumptions in the code that can lead
to unexpected results.

Assertions catch these assumptions and stop the program.

This allows us to understand why incorrect results happen
more easily.

jannaleong committed:
Merge pull request #1 from jannaleong/A-CodeQuality

Improve Code Quality

jannaleong committed:
Merge branch 'master' into A-Assertions

# Conflicts:
#	src/main/data/belleList.txt

jannaleong committed:
Merge tag 'A-Assertions'

* tag 'A-Assertions':
  Add assertions to code

# Conflicts:
#	src/main/data/belleList.txt

jannaleong committed:
Merge pull request #2 from jannaleong/A-Assertions

Add assertions to code

jannaleong committed:
Update to belle.txt

jannaleong committed:
Merge branch 'master' of https://github.com/jannaleong/ip

* 'master' of https://github.com/jannaleong/ip:

# Conflicts:
#	src/main/data/belleList.txt

jannaleong committed:
Add SnoozeCommand

Users are unable to change the deadlines of tasks once they are set.

SnoozeCommand allows users to extend their deadline.

This allows for a more flexible planner.

dabzpengu commented on src/main/java/duke/run/Parser.java:
```diff
@@ -0,0 +1,58 @@
+     * @throws BelleException  If an invalid belle.belle.command is entered.
+     */
+    public Command parse(String input) throws BelleException {
+        String[] inputlist = input.split(" ");
```
variable name inputlist should be in camelCase, inputList

dabzpengu commented on src/main/java/duke/run/Storage.java:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @param filepath File containing storage data.
+     */
+    public Storage(String filepath) {
```
variable name filepath should be in camelCase, filePath

dabzpengu commented on src/main/java/duke/tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,38 @@
+    /**
+     * Constructs DeadlineTask.
+     */
+    public DeadlineTask(String name, boolean done, String deadline) {
```
boolean type variable name done can be isDone for further clarity

jannaleong committed:
Add name to GUI title bar

Title bar has no title.

Name of GUI is unknown which is confusing.

Add name to title bar to confirm
name of product.

jannaleong committed:
Add user guide

With no user guide, new users will be confused
as to how to use Belle.

Add user guide with all basic commands and expected outputs.

jannaleong committed:
Adjust code to allow user to exit program with bye

Code allows user to exit program only after entering
bye command followed by another input.

This is not very efficient.

Adjusted code so program immediately exits after
bye command is inout in.

jannaleong committed:
Add more javadocs

Some methods to do not have javadocs.

This creates confusion on what the method does.

Add javadocs for clarity on this.

jannaleong committed:
Improve code quality

Code is generally of decent quality
but there is room for improvement.

These improvements help to make
the code more readable and easy
to understand.

jannaleong committed:
Add more assert statements

Some significant assumptions were
not checked.

Add more assertions to ensure
assumptions are correct.

jannaleong committed:
Add more javadocs

Some methods had no javadocs or
missed out parameters.

Improve upon javadocs so user
is clearer on what the methods and
classes do.

jannaleong committed:
Add junit test for tasklist class

Manually tested using user commands.

Junit testing makes testing this class
more efficient.

jannaleong committed:
Improve on coding standard

Mostly followed coding standard.

There are certain parts that could
be improved.

Code is more readable.

jannaleong committed:
Improve user guide

User guide does not contain
screenshot or some formatting.

With additions, it is more
readable and user friendly.

jannaleong committed:
Improve coding quality

Code does not have spaces
to section different kinds of logic.

Add spaces so logic is clearer.

jannaleong committed:
Make jar file os independent

Jar file only works on mac.

Jar file now works on mac
and windows.

jannaleong committed:
Abstract execute method in snooze

Method is too long and hard to read.

Abstract into another method in
task to make it more readable.

jannaleong committed:
Fix variable name

Change variable name to something
more specific. This makes code easier
to understand.

jannaleong committed:
Remove duke from all source code

There is a mix of belle and duke
in source code.

This is confusing.

Change all duke to belle.

jannaleong committed:
Fix bye bug

Bye print message does not
display before program exits.

With a delay, bye message
can be seen.

jannaleong committed:
Fix date bug

With snooze command, date
isnt formatted.

Change to make date in the correct
format for deadline and event tasks.

jannaleong committed:
Improve code quality

Small fixes to align more with
coding standard and quality.

jannaleong committed:
Fix snooze bug

Format of date is not correct.

Adjust code so date is more
readable.

jannaleong committed:
Specify snooze function

In readme, snooze function is
not very clear.

Make snooze function clearer
for event task.

jannaleong committed:
Fix list bug

List would not show all
tasks as there is not enough
space.

Set prefix height to show
all tasks.
